src/platform/Darwin/system/SystemLayerImplDispatchSockets.mm

```diff
@@ -0,0 +1,293 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements Layer using dispatch and sockets.
+ */
+
+#if !__has_feature(objc_arc)
+#error This file must be compiled with ARC. Use -fobjc-arc flag (or convert project to ARC).
+#endif
+
+#include <system/SystemLayerImplDispatch.h>
+
+#include <lib/support/CodeUtils.h>
+
+namespace chip {
+namespace System {
+    CHIP_ERROR LayerImplDispatch::StartWatchingSocket(int fd, SocketWatchToken * tokenOut)
+    {
+        SocketWatch * watch = nullptr;
+
+        ReturnErrorOnFailure(mSocketWatchPool.Allocate(fd, watch));
+        *tokenOut = reinterpret_cast<SocketWatchToken>(watch);
+
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR LayerImplDispatch::StopWatchingSocket(SocketWatchToken * tokenInOut)
+    {
+        __auto_type * watch = reinterpret_cast<SocketWatch *>(*tokenInOut);
+
+        ReturnErrorOnFailure(mSocketWatchPool.Release(watch));
+        *tokenInOut = InvalidSocketWatchToken();
+
+        return CHIP_NO_ERROR;
```

Comment:
SystemLayerImplSelect signals the thread to stop selecting. I guess we don't need to because we never select?